docs/Homebrew-and-Java.md

```diff
@@ -0,0 +1,40 @@
+# Homebrew and Java
+
+This page describes how Java is handled in Homebrew for users. Prospective formula authors may refer to existing Java-based formulae for examples of how to install packages written in Java via Homebrew, or visit the [Homebrew discussion forum](https://github.com/Homebrew/discussions/discussions) to seek guidance for their specific situation.
+
+Most Java-based packages in Homebrew use the default Homebrew-provided `openjdk` package for their JDK dependency, although some packages with specific version requirements may use a versioned package such as `openjdk@8`.
+
+## Using Homebrew's JDKs as the macOS default
+
+For many years, Apple has not shipped a default JDK as part of macOS. Instead, attempting to run the `java` command will result in an error as follows:
+
+```console
+% java
+The operation couldn’t be completed. Unable to locate a Java Runtime.
+Please visit http://www.java.com for information on installing Java.
+```
+
+Similarly, when double-clicking in a JAR file in Finder, an error message is shown directing the user to install a Java runtime:
+
+<img src="assets/img/docs/java-macos-finder-error.png" width="372" alt="Screen shot of Finder error">
+
+For the system Java wrappers to find the JDK provided by the `openjdk` formula, symlink it with:
+
+```console
+% sudo ln -sfn /usr/local/opt/openjdk/libexec/openjdk.jdk /Library/Java/JavaVirtualMachines/openjdk.jdk
+```
+
+To confirm that your JDK has been correctly linked by this command, you can use the `/usr/libexec/java_home` command:
+
+```console
+% /usr/libexec/java_home -V
+Matching Java Virtual Machines (1):
+    15.0.1 (x86_64) "UNDEFINED" - "OpenJDK 15.0.1" /usr/local/Cellar/openjdk/15.0.1/libexec/openjdk.jdk/Contents/Home
+/usr/local/Cellar/openjdk/15.0.1/libexec/openjdk.jdk/Contents/Home
+```
+
+Other versioned JDK formulae have their own instructions that can be viewed with `brew info`. More information about controlling the system JDK installation can be found by viewing `man 1 java_home`.
```

Comment:
I'd rather we didn't recommend any of this. It's not relevant to Homebrew specifically.

Reply:
I added these docs at @jonchang's recommendation, so let's let Jon chime in maybe?

Given that I've already spent a few hours on this and I don't think I have much more to add, would you guys mind taking it from here? The PR is set up to allow maintainers to edit it, so you should be able to remove anything that doesn't fit with your goals for these docs.

Reply:
@md5 No word from Jon so I'd rather this was removed, thanks.

Reply:
To be clear what you're asking, do you want the whole "Using Homebrew's JDKs as the macOS default" section removed? The part you highlighted doesn't include the whole section, just how to solve the problem. The unhighlighted part of the section still describes the problem itself, which seems unhelpful on its own from my perspective.

Reply:
> do you want the whole "Using Homebrew's JDKs as the macOS default" section removed?

Yes, please.

Reply:
Done in e63bbf46fd8d1b3ad6a5983dbc7be390aaf14d46

Reply:
Ended up squashing this down to one commit since I had added a binary for the screenshot that doesn't need to stay in the history, so the amended commit without that section is now e92d67d15c9d00bce05504dedc52491c8be71925